script.js

```diff
@@ -1,61 +1,38 @@
-function loadDataFromStorage() {
-  let list = JSON.parse(window.localStorage.getItem('bookshelf'));
-  if (!list) {
-    // If it is empty then fill some data to work
-    list = [
-      {
-        id: 0,
-        title: 'Book 1',
-        author: 'Test test',
-      },
-      {
-        id: 1,
-        title: 'Book 2',
-        author: 'Test test',
-      },
-      {
-        id: 2,
-        title: 'Book 3',
-        author: 'Test test',
-      },
-      {
-        id: 3,
-        title: 'Book 4',
-        author: 'Test test',
-      },
-    ];
+class BookShelf {
+  constructor() {
+    this.#loadDataFromStorage();
   }
-  return list;
-}
 
-function saveData(bookList) {
-  window.localStorage.setItem('bookshelf', JSON.stringify(bookList));
-}
+  #loadDataFromStorage() {
+    this.bookList = JSON.parse(window.localStorage.getItem('bookshelf'));
+    if (!this.bookList) this.bookList = [];
+  }
 
-function addBook(title, author, bookList) {
-  let index = 0;
-  if (bookList.length !== 0) {
-    index = bookList[bookList.length - 1].id + 1;
+  saveData() {
+    window.localStorage.setItem('bookshelf', JSON.stringify(this.bookList));
   }
-  bookList.push({
-    id: index,
-    title,
-    author,
-  });
-}
 
-function BookShelf() {
-  this.bookList = loadDataFromStorage();
-  // Add functions to delete and to add a book
-  this.removeBook = (id) => {
+  addBook(title, author) {
+    let index = 0;
+    if (this.bookList.length !== 0) {
+      index = this.bookList[this.bookList.length - 1].id + 1;
+    }
+    this.bookList.push({
```

Comment:
- [ ] **[OPTIONAL]** It would be nice to use the `concat` here instead of `push` to avoid mutating the array.